---
layout: index
published: true
---

> Viromics tutorial for the [**10th EBAME workshop**](https://maignienlab.gitlab.io/ebame/)
> with Evelien Adriaenssens and Andrea Telatin

In this tutorial we will mine for phage sequences in a metagenome assembly.

Checking their quality, we can then select some of the predictions as candidate 
"Viral [OTUs](https://en.wikipedia.org/wiki/Operational_taxonomic_unit)" (vOTUs).

When you have multiple samples, you can **back-map the reads** to the vOTUs to estimate their abundance in the original samples.

Using **Anvi'o**, we can generate a visualization focussed on the vOTUs, so building a CONTIGS.db database only based on the candidate vOTUs.

There are also some optional sections, covering how to automate and executing **workflows** written in NextFlow.

---

Tutorial developed by:
 * Evelien Adriaenssens
 * Andrea Telatin
 * Alise Ponsero
 * Ryan Cook
 * Claire Elek
 * Hannah Pye

from the [Quadram Institute Bioscience](https://quadram.ac.uk/), Norwich, UK.
