---
title: "Before we start"
---

## Prerequisites

* Basic understanding of microbiology and microbial genomics
* Knowledge of operating with the Linux Command Line Interface (CLI), understanding the file paths and using common commands like `ls`, `cd`, `head`, `wc` ...
* Ability to install packages and create/activate environments using conda (or modern alternatives such as `mamba` or `micromamba`)
* *Optional*: ability to write simple bash scripts, perform `for` loops, using Bash variables (like `$REFERENCE`)


### Suggested readings

- **"[Metagenomic approaches in microbial ecology: an update on whole-genome and marker gene sequencing analysis](https://pmc.ncbi.nlm.nih.gov/articles/PMC7641418/)" (2020, PMC7641418)**  
  This relatively recent review offers a detailed introduction to metagenomics with a strong emphasis on analysis pipelines for both whole-genome and marker gene data, covering best practices, comparisons of tools, and guidelines[1]

- **"[Introduction to Metagenomics: Tools, Workflows & Insights](https://www.rna-seqblog.com/introduction-to-metagenomics-tools-workflows-insights/)" (2025, RNA-seq Blog)**  
  A good blog post with a broad overview that bridges conceptual foundations with practical workflows, describing sequencing technologies, data analysis strategies, and examples of ecological discoveries[2]

- **"[Genome-resolved metagenomics: a game changer for microbiome studies](https://www.nature.com/articles/s12276-024-01262-7)" (2024, Nature Cell Biology)**  
  This review explains the transition from traditional methods to genome-resolved metagenomics, highlighting its impact on human microbiome research with approachable explanations for students.[3]

- **"Metagenomics - a guide from sampling to data analysis" (2012, PMC3351745)**  
  An accessible primer (a bit outdated), this article provides a clear step-by-step guide from sampling through data analysis, discussing both functional and phylogenetic aspects with practical examples.[4]
 
- **"[Metagenomics: An Effective Approach for Exploring Microbial Diversity](https://pmc.ncbi.nlm.nih.gov/articles/PMC10252221/)" (2023, PMC10252221)**  
  Focused on applications in human microbiome research, this review is easy to follow and discusses technological advances, pipelines, and key databases relevant for analysis.[6]

- **[A practical guide to amplicon and metagenomic analysis of microbiome data](https://pmc.ncbi.nlm.nih.gov/articles/PMC8106563/)** (2024)
  This guide shows how 16S (amplicon) and metagenomics workflow produce their data and how they can be integrated with downstream analyses[7]

 